src/overlays/actors/ovl_En_Fhg_Fire/z_en_fhg_fire.c

```diff
@@ -21,27 +45,492 @@ const ActorInit En_Fhg_Fire_InitVars = {
+        this->collider.dim.radius = 40;
+        this->collider.dim.height = 50;
+        this->collider.dim.yShift = -25;
+        this->light = Lights_Insert(globalCtx, &globalCtx->lightCtx, (void*)(&this->unk_1A0));
+        Lights_InitType0PositionalLight(&this->unk_1A0, thisx->posRot.pos.x, thisx->posRot.pos.y, thisx->posRot.pos.z,
+                                        0xFF, 0xFF, 0xFF, 0xFF);
+    }
+}
+
+void EnFhgFire_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnFhgFire* this = THIS;
+
+    if ((thisx->params == 0x23) || (thisx->params == 0x24) || (thisx->params == 0x32)) {
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+
+    if (thisx->params == 0x32) {
+        Lights_Remove(globalCtx, &globalCtx->lightCtx, this->light);
+    }
+}
+
+void func_80A0F6F8(EnFhgFire* this, GlobalContext* globalCtx) {
+    Camera* camera;
+    s32 pad;
+    Vec3f randVec;
+    Vec3f tmpVec;
+    s16 i;
+    s16 randY;
+    s16* tmp;
+
+    tmp = &this->unk_156;
+    camera = Gameplay_GetCamera(globalCtx, 0);
+
+    switch (this->fireMode) {
+        case 0x00:
+            this->fireMode = 0x0A;
+            this->unk_150.x = 0x07;
+            break;
+
+        case 0x0A:
+            this->actor.shape.rot.y = func_8005A948(camera) + ((*tmp & 0xFF) << 0x0F);
+            Math_SmoothScaleMaxF(&this->scale, 1.0f, 1.0f, 0.2f);
+
+            if (this->unk_150.x == 0) {
+                this->fireMode = 0x0B;
+                this->actor.shape.rot.z += 0x8000;
+                this->unk_150.x = 0x25;
+                this->actor.posRot.pos.y -= 200.0f;
+
+                Actor_SpawnAttached(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_FHG_FIRE,
+                                    this->actor.posRot.pos.x, this->actor.posRot.pos.y, this->actor.posRot.pos.z, 500,
+                                    0, 0, 0x24);
+
+                tmpVec = D_80A117BC;
+
+                for (i = 0; i < 35; i++) {
+                    randVec.x = Math_Rand_CenteredFloat(30.f);
+                    randVec.y = Math_Rand_ZeroFloat(5.0f) + 3.0f;
+                    randVec.z = Math_Rand_CenteredFloat(30.f);
+                    tmpVec.y = -0.2f;
+                    EffectSsFhgFlash_Spawn(globalCtx, &this->actor.posRot.pos, &randVec, &tmpVec,
+                                           (s16)(Math_Rand_ZeroOne() * 100.0f) + 240, 0);
+                }
+
+                func_80033E88(&this->actor, globalCtx, 4, 10);
+            }
+
+            break;
+
+        case 0x0B:
+            this->actor.shape.rot.y = func_8005A948(camera) + ((*tmp & 0xFF) << 0x0F);
+
+            Math_SmoothScaleMaxF(&this->scale, 0.0f, 1.0f, 0.2f);
+            if (this->unk_150.x == 0x1E) {
+                randY = (Math_Rand_ZeroOne() < 0.5f) ? 0x1000 : 0;
+
+                for (i = 0; i < 8; i++) {
+                    if (1) {
+                        do {
+                        } while (0);
```

Comment:
this can all be on one line, helps to avoid as much clutter as possible when we do dumb stuff like this to match

Reply:
just noticed all that stuff is in the if(1). sometimes you can get away with just keeping it on one line anyway even if this is what the permuter suggested. try it out

Reply:
Even better : this is no longer needed to match.